# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/PresShell.cpp
# Commit: 3baead3e079b
# Full Hash: 3baead3e079bc90e7637f7f338a185149a58aa3e
# Author: Miko Mynttinen <mikokm@gmail.com>
# Date: 2022-02-22 09:37:09
# Regressor Bug: 1714584
# File Overlap Count: 1
# Description:
#   Bug 1714584 - Part 2: Remove nsDisplayList::RemoveBottom() r=mstange
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D138153
# ==============================================================================

diff -r a2da895a58ce -r 3baead3e079b layout/base/PresShell.cpp
--- a/layout/base/PresShell.cpp	Tue Feb 22 00:44:25 2022 +0000
+++ b/layout/base/PresShell.cpp	Tue Feb 22 00:44:25 2022 +0000
@@ -4692,12 +4692,10 @@
   // part of the selection. Then, append the wrapper to the top of the list.
   // Otherwise, just delete the item and don't append it.
   nsRect surfaceRect;
-  nsDisplayList tmpList;
-
-  nsDisplayItem* i;
-  while ((i = aList->RemoveBottom())) {
+
+  for (nsDisplayItem* i : aList->TakeItems()) {
     if (i->GetType() == DisplayItemType::TYPE_CONTAINER) {
-      tmpList.AppendToTop(i);
+      aList->AppendToTop(i);
       surfaceRect.UnionRect(
           surfaceRect, ClipListToRange(aBuilder, i->GetChildren(), aRange));
       continue;
@@ -4778,7 +4776,7 @@
     // list, insert that as well
     nsDisplayList* sublist = i->GetSameCoordinateSystemChildren();
     if (itemToInsert || sublist) {
-      tmpList.AppendToTop(itemToInsert ? itemToInsert : i);
+      aList->AppendToTop(itemToInsert ? itemToInsert : i);
       // if the item is a list, iterate over it as well
       if (sublist)
         surfaceRect.UnionRect(surfaceRect,
@@ -4789,9 +4787,6 @@
     }
   }
 
-  // now add all the items back onto the original list again
-  aList->AppendToTop(&tmpList);
-
   return surfaceRect;
 }
 